From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Chosen <haizhongdelan@outlook.com>
Date: Sat, 15 Apr 2023 12:53:09 +0800
Subject: [PATCH] Add API


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 1d1a1d087dabc9794e0062a064da2cced4062309..da7d2a1aaf8f8f664b8fe11e1970560d35e9b583 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1994,6 +1994,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // Skydom start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getSkydomConfig()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Skydom end
         /**
          * Sends the component to the player
          *
