From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Chosen <haizhongdelan@outlook.com>
Date: Sat, 15 Apr 2023 12:53:09 +0800
Subject: [PATCH] Add API


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 5b225bbb128893d67251a96ab318035802a0cf76..e75ce6ec2fbbfed17d2c76348bd0e00674a38c56 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2007,6 +2007,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // Skydom start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getSkydomConfig()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Skydom end
         /**
          * Sends the component to the player
          *
