From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Chosen <haizhongdelan@outlook.com>
Date: Sat, 15 Apr 2023 12:53:09 +0800
Subject: [PATCH] Add API


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 9f7029d327b55c8b1b9cfef8ea2678b87dad8ddb..bbccef5a4e17a008cb6929b03766c1671fa0a99a 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1967,6 +1967,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // Skydom start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getSkydomConfig()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Skydom end
         /**
          * Sends the component to the player
          *
