From: Chosen <haizhongdelan@outlook.com>
Date: Sun, 2 Apr 2023 17:18:59 +0800
Subject: [PATCH] Cancel protect destroy

diff --git a/src/main/java/net/minecraft/world/level/portal/PortalForcer.java b/src/main/java/net/minecraft/world/level/portal/PortalForcer.java
index d1c7eba29c64a6dbf55d3062ced9769eecb548ed..faaf50cb9bce254aef554ed8b402b145532e12a4 100644
--- a/src/main/java/net/minecraft/world/level/portal/PortalForcer.java
+++ b/src/main/java/net/minecraft/world/level/portal/PortalForcer.java
@@ -241,18 +241,12 @@ 
        for (int j = -1; j < 3; ++j) {
            for (int k = -1; k < 4; ++k) {
                temp.setWithOffset(pos, portalDirection.getStepX() * j + enumdirection1.getStepX() * distanceOrthogonalToPortal, k, portalDirection.getStepZ() * j + enumdirection1.getStepZ() * distanceOrthogonalToPortal);
-                // Paper start - prevent destroying unbreakable blocks
-                if (!io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.allowPermanentBlockBreakExploits) {
-                    if (!this.level.getBlockState(temp).isDestroyable()) {
-                        return false;
-                    }
-                }
-                // Paper end - prevent destroying unbreakable blocks
+                // Cancel protect destroy by Skydom
                if (k < 0 && !this.level.getBlockState(temp).getMaterial().isSolid()) {
                    return false;
                }

                if (k >= 0 && !this.canPortalReplaceBlock(temp)) {
                    return false;
                }
            }
        }